From 8747e96db58a9c18e17861d4d8ce697a0b14fada Mon Sep 17 00:00:00 2001
From: Neal Gompa <ngompa13@gmail.com>
Date: Sun, 17 Sep 2017 09:34:57 +0000
Subject: [PATCH] ext: Ignore DistEpoch entirely

For now, we don't particularly care about DistEpoch, and would
rather ignore it, since DNF can't use it right now.
---
 ext/repo_mdk.c   | 6 ++++--
 ext/repo_rpmdb.c | 2 ++
 ext/repo_rpmmd.c | 6 ++++++
 3 files changed, 12 insertions(+), 2 deletions(-)

diff --git a/ext/repo_mdk.c b/ext/repo_mdk.c
index 345d416..cb8be6f 100644
--- a/ext/repo_mdk.c
+++ b/ext/repo_mdk.c
@@ -183,15 +183,17 @@ repo_add_mdk(Repo *repo, FILE *fp, int flags)
 	      if (n && !strncmp(n + 1, disttag, strlen(disttag)))
 		*n = 0;
 	    }
+	  /* // Ignore DistEpoch entirely
 	  if (distepoch && *distepoch)
 	    {
-	      /* add distepoch */
+	      // add distepoch
 	      int le = strlen(distepoch);
 	      int ln = strlen(nvra);
 	      nvra[ln++] = ':';
-	      memmove(nvra + ln, distepoch, le);	/* may overlap */
+	      memmove(nvra + ln, distepoch, le);	// may overlap
 	      nvra[le + ln] = 0;
 	    }
+	   */
 	  version = strrchr(nvra, '-');
 	  if (version)
 	    {
diff --git a/ext/repo_rpmdb.c b/ext/repo_rpmdb.c
index c339f62..9cc9b82 100644
--- a/ext/repo_rpmdb.c
+++ b/ext/repo_rpmdb.c
@@ -384,6 +384,7 @@ static char *headtoevr(RpmHead *h)
       sprintf(evr, "%s-%s", version, release);
     }
   distepoch = headstring(h, TAG_DISTEPOCH);
+  /* //Ignore DistEpoch entirely
   if (distepoch && *distepoch)
     {
       int l = strlen(evr);
@@ -391,6 +392,7 @@ static char *headtoevr(RpmHead *h)
       evr[l++] = ':';
       strcpy(evr + l, distepoch);
     }
+   */
   return evr;
 }
 
diff --git a/ext/repo_rpmmd.c b/ext/repo_rpmmd.c
index 729f4f7..ccc4d9f 100644
--- a/ext/repo_rpmmd.c
+++ b/ext/repo_rpmmd.c
@@ -414,6 +414,12 @@ makeevr_atts(Pool *pool, struct parsedata *pd, const char **atts)
   if (r)
     {
       *c++ = '-';
+      const char *rde = strchr(r, ':');
+      /* Drop DistEpoch from 'rel' */
+      if (rde != 0)
+        {
+            *rde = '\0';
+        }
       strcpy(c, r);
       c += strlen(c);
     }
-- 
2.14.1

